### ISSUE TITLE
Specify amazon-ebs builder instance kms key for encryption

### ISSUE THREAD

**Issue by atheiman** [hashicorp/packer#8144]
I want to be able to:

- [default my account ebs encryption to aws-managed kms key (`aws/ebs`)](https://aws.amazon.com/blogs/aws/new-opt-in-to-default-encryption-for-new-ebs-volumes/)
- build an ami using `amazon-ebs` builder, the resulting ami must be encrypted with a customer-managed kms key in the account - not with the aws-managed `aws/ebs` key
  - *this is what I can't figure out*
- share the resulting ami with another account
  - I think this can be done with snapshot_users as of a few months ago, though I am just doing it after builds using awscli

I think there are two ways for this to work:
- allow specifying the kms key to be used on the builder instance, resulting in an ami encrypted with the customer-managed kms key (better)
- encrypt the builder instance with the default kms key for the account, then copy the resulting ami and re-encrypt with a specified customer-managed kms key (worse)

Am I going about this all wrong? I've tried many combinations of settings in my template, but I cannot get an ami encrypted with my customer-managed kms key. Am I the only one that is defaulting my account to ebs encryption using `aws/ebs` key?

**Comment by atheiman** [comment]
My source ami for this build was encrypted with the aws-managed key `aws/ebs`. Even after switching my account default ebs encryption to my customer-managed key, and my resulting ami was still encrypted with `aws/ebs`

**Comment by atheiman** [comment]
I just turned off account default encryption, and specified an unencrypted ami as my source, and then `kms_key_id` did have the intended effect.

So I guess the real issue here is, if source ami is encrypted the builder instance will be encrypted with that same key no matter what? I dont think that is a limitation of EC2.

I also noticed my builder did not have the correct `kms:CreateGrant` permissions part of the way thru messing with this, so that may have been a root cause for my issues.
